### ISSUE TITLE
ColumnDataSource update duplicate graph 3.5[BUG] 

### ISSUE THREAD

**Issue by AlbanWaxin** [bokeh/bokeh#13983]
### Software versions

Python version        :  3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
IPython version       :  (not installed)
Tornado version       :  6.4.1
Bokeh version         :  3.5.0
BokehJS static path   :  ###########\.venv\Lib\site-packages\bokeh\server\static
node.js version       :  (not installed)
npm version           :  (not installed)
jupyter_bokeh version :  (not installed)
Operating system      :  Windows-10-10.0.19045-SP0

### Browser name and version

Firefox

### Expected behavior

_Note: this issue is related to bokeh discourse post : [link](https://discourse.bokeh.org/t/3-5-update-change-of-behaviour-for-streaming-data/11699)_

I am currently building a system that represents live data in four plots. My system uses the Bokeh server to update the contents of a data source as new data is read from a growing local file.

The four plots consist of two color-mapped images and two line graphs. I started this project with Bokeh 3.4.1

Every update ( with `add_periodic_callback`) the data source of my figure is updated and my image should shift to the left

![image](https://github.com/user-attachments/assets/7120139f-5bbc-443d-b7c5-77f9efa58c0a)

### Observed behavior

In Bokeh 3.5.0 the same code give another behaviour.

Every time my data is updated (a change in ColumnDataSource), the graph does not update but instead creates a smaller version of itself in the same visual space.

![image](https://github.com/user-attachments/assets/7f5e417b-2c85-4843-b99f-a84e348483a3)

It seems that this behaviour only appeared on firefox

### Example code

```Python
#I can't communicate the whole code has it is blended in my code base but here are the main part concerning those object.

class Spectrogram():
    def __init__(self,parameters:Spectrogram_Params,):
        print(parameters)
        #================================================================
        # Display Parameters
        #================================================================
        self.color_mapper = parameters.color_mapper
        self.color_bar = parameters.colorbar
        #================================================================
        # Stored Data
        #================================================================
        self.data_buffers = {
            "timestamps": [],
            "freqlist": np.empty(0),
        }
        #================================================================
        # Plot Tools
        #================================================================
        self.tools = []
        for tool in parameters.tools:
            match tool[0]:
                case "pan":
                    self.tools.append(PanTool())
                case "boxzoom":
                    self.tools.append(BoxZoomTool())
                case "wheelzoom":
                    self.tools.append(WheelZoomTool())
                case "save":
                    self.tools.append(SaveTool())
                case "hover":
                    self.tools.append(HoverTool(tooltips=parameters.tooltips,formatters={"$x":"datetime"}))
                case "tap":
                    self.tools.append(TapTool())
                case "reset":
                    self.tools.append(ResetTool())
                case _:
                    print("Unknown tool : ",tool[0])
            self.tools[-1].visible = tool[1]
        self.timestamp_formatter = DatetimeTickFormatter(hourmin='%H:%M:%S',minsec='%H:%M:%S',milliseconds='%H:%M:%S',seconds='%H:%M:%S',minutes='%H:%M:%S',days='%H:%M:%S',months='%H:%M:%S',years='%H:%M:%S')
        #================================================================
        # Plot Components
        #================================================================
        # Spectrogram
        self.source = ColumnDataSource(data={'image':[[]],'x':[[]],'y':[[]],'dw':[[]],'dh':[[]]})
        self.figure = figure(title=parameters.title,x_axis_label=parameters.xaxis.label,y_axis_label=parameters.yaxis.label,x_axis_type=parameters.xaxis.data_type,y_axis_type=parameters.yaxis.data_type,sizing_mode='stretch_width',toolbar_location='above',tools=self.tools,)
        if parameters.xaxis.temporal:
            self.figure.xaxis[0].formatter = self.timestamp_formatter
        self.figure.x_range.range_padding = 0
        self.figure.y_range.range_padding = 0
        self.figure.toolbar.logo = None # type: ignore
        # Reticle
        self.source_reticle = ColumnDataSource(data={'hx':[[0,0]],'hy':[[0,0]],'vx':[[0,0]],'vy':[[0,0]]})
        #================================================================
        # Plot state
        #================================================================
        self.reticle_displayed = 0


def update_data(self,timestamps:list|np.ndarray,matrix:np.ndarray,freqlist:np.ndarray,length:int=2048):
        #Replace buffers if necessary
        #print(len(timestamps),matrix.shape,freqlist,flush=True)
        print(timestamps,matrix,freqlist,length,flush=True)
        if (freqlist != np.linspace(0,100,length)).all() or len(self.data_buffers['freqlist']) == 0:
            self.data_buffers['freqlist'] = freqlist
        self.data_buffers['timestamps'] = timestamps

        # Update data sources
        self.source.data.update(x =[timestamps[0]],dw=[timestamps[-1]-timestamps[0]],y=[self.data_buffers['freqlist'][0]],dh=[self.data_buffers['freqlist'][-1] - self.data_buffers['freqlist'][0]],image=[matrix])
        self.update_reticle()
```

### Stack traceback or browser console output

It also seems important to say that these figures are displayed in a row and column layout and added as the root in the curdoc element.

No error is reported in the bokeh server console output.

**Comment by mattpap** [comment]
There should be errors and/or exceptions reported in browser's developers console (Ctrl + Shift + J or Ctrl + Shift + I).

**Comment by AlbanWaxin** [comment]
```
Error rendering Bokeh items: Error: invalid bbox {x: 103.49777221679688, y: 605, width: 41, height: NaN}
    _ http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:230
    _rect http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:346
    rect http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:346
    bbox http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:346
    k http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:355
    _draw_oriented_labels http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:355
    _draw_major_labels http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:355
    _paint http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:355
    paint http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:249
    _paint_levels http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:616
    _actual_paint http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:616
    paint http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:616
    _after_layout http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:616
    after_layout http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:581
    after_layout http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:581
    after_layout http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:581
    compute_layout http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:581
    invalidate_layout http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:581
    _after_render http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:581
    after_render http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:277
    r_after_render http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:223
    build http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:223
    l http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:220
[bokeh.min.js:164:978](http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d)
    k http://localhost:5006/static/js/bokeh.min.js?v=4885298c46175e086cef50e41170e1d47cb77faa4020ef403fb6a65d313bcc55ee11e012be061d7dc6affb7fe3e44e03140a12dce87566a478efea820562e68d:164
```
    
There is only this error that is repeated at every update

**Comment by mattpap** [comment]
What version of Firefox do you use? This looks to me like a duplicate of issue #13969.

**Comment by AlbanWaxin** [comment]
My firefox is in version 115.13.0esr (64 bits) which according to firefox is the last i can get

**Comment by mattpap** [comment]
So this is indeed issue #13969. This will be fixed in bokeh 3.5.1 (https://github.com/bokeh/bokeh/pull/13982), which will be released later this week.
